### PR TITLE
test: move clientconn state transition test to test/ directory

### PR DESCRIPTION
**Pull request by easwars**
I'm making some other test changes which introduce a dependency on the `grpc` package for `internal/testutils`. Currently the clientconn state transition tests introduce a dependency from package `grpc` to `internal/testutils`. This leads to an import cycle. This PR removes the latter dependency by moving these tests out to the `test` directory. The added benefit of doing this is that we are forced to use the public API instead of relying on some internal APIs.

This also introduces a little bit of code duplication, as some of the test helper functions need to be repeated for tests in `grpc` package and for the `test` package. This can be removed in the future when we switch to using the public API  more frequently in our tests.

RELEASE NOTES: n/a

### CODE REVIEW

File: test/clientconn_state_transition_test.go [hunk 1]
```diff
@@ -35,10 +37,7 @@ import (
 	"google.golang.org/grpc/resolver/manual"
 )
 
-const (
-	stateRecordingBalancerName = "state_recoding_balancer"
-	defaultTestTimeout         = 10 * time.Second
-)
+const stateRecordingBalancerName = "state_recoding_balancer"
```
Comment (zasweq) [inline test/clientconn_state_transition_test.go:40]: Here as well.

File: test/clientconn_state_transition_test.go [hunk 2]
```diff
@@ -498,3 +502,20 @@ func keepReading(conn net.Conn) {
 	for _, err := conn.Read(buf); err == nil; _, err = conn.Read(buf) {
 	}
 }
+
+// stayConnected makes cc stay connected by repeatedly calling cc.Connect()
+// until the state becomes Shutdown or until 10 seconds elapses.
```
Comment (zasweq) [inline test/clientconn_state_transition_test.go:507]: This comment no longer applies as the previous helper did, as the 10 seconds is no longer hardcoded. Switch to context expires.

**Review by zasweq — APPROVED** [review]
LGTM outside minor comments.

File: clientconn_test.go [hunk 3]
```diff
@@ -44,6 +46,17 @@ import (
 	"google.golang.org/grpc/testdata"
 )
 
+const (
+	defaultTestTimeout         = 10 * time.Second
+	stateRecordingBalancerName = "state_recoding_balancer"
```
[the remark this answers is not in the document]
Comment (easwars) [reply clientconn_test.go]: wow .. what eyes ... done

File: test/clientconn_state_transition_test.go (hunk 1 above)
Comment (easwars) [reply test/clientconn_state_transition_test.go]: done

File: test/clientconn_state_transition_test.go (hunk 2 above)
Comment (easwars) [reply test/clientconn_state_transition_test.go]: done, thanks

**Pull request merged**